src/expr/impl/src/udf/quickjs.rs

```diff
@@ -32,47 +33,68 @@ static QUICKJS: UdfImplDescriptor = UdfImplDescriptor {
         })
     },
     build_fn: |opts| {
-        let mut runtime = Runtime::new()?;
-        if opts.kind.is_aggregate() {
-            runtime.add_aggregate(
-                opts.identifier,
-                Field::new("state", DataType::Binary, true).with_metadata(
-                    [("ARROW:extension:name".into(), "arrowudf.json".into())].into(),
-                ),
-                UdfArrowConvert::default().to_arrow_field("", opts.return_type)?,
-                CallMode::CalledOnNullInput,
-                opts.body.context("body is required")?,
-            )?;
-        } else {
-            let res = runtime.add_function(
-                opts.identifier,
-                UdfArrowConvert::default().to_arrow_field("", opts.return_type)?,
-                CallMode::CalledOnNullInput,
-                opts.body.context("body is required")?,
-            );
+        // NOTE: Some function calls such as `add_function()` requires async.
+        // However, since the `Runtime` here is not shared, the async block will never block.
+        futures::executor::block_on(async {
```

Comment:
What about `now_or_never().expect()`?

Reply:
IIUC, it can work only if the internal implementation of these async functions return a ready promise. I don't want to rely on such assumption.